### ISSUE TITLE
Postman OK, Newman error: Unexpected token u in JSON at position 0 on globals variable?

### ISSUE THREAD

**Issue by patrickgardella** [postmanlabs/newman#1457]
1. Newman 3.9.3, Postman 6.0.8
2. Windows 7 Enterprise SP1 x64 (win32 6.1.7601 / x64)
3. Running via CLI
4. This has worked until very recently. It still works in TFS for the last globals file which was checked in on 2/19. I do not know which version of Newman is running on the TFS build servers.
5. The test should complete with a 2 valid tests, as it does in Postman
6. Command used:

`newman run --global tests\PostmanEnvironments\globals.postman_globals.json --environment tests\PostmanEnvironments\VisualStudio.postman_environment.json --color --insecure --reporters cli,junit --reporter-junit-export results\junit.xml tests\CareOptimizationServices.postman_collection.json --bail`

7. I can send the collection, globals, and environment files via DM if needed. The API is local, so you won't be able to run them.

8. Screenshots:

Successful run in Postman:
![postman](https://user-images.githubusercontent.com/6595897/36994147-289005b4-207e-11e8-9448-5eb22f7aa24d.png)

Failed run in Newman
![newman](https://user-images.githubusercontent.com/6595897/36993880-61e6b02a-207d-11e8-8adf-f4b5cda6abb9.PNG)

Successful run in Newman on our TFS Server during a build:
```
2018-02-28T20:49:24.6609452Z Folder Authorization
2018-02-28T20:49:24.6609452Z Sub-folder api/authorization/login
2018-02-28T20:49:38.2235578Z   POST http://test-services-TA.Humana.com/COSAPI/api/Authorization [200 OK, 956B, 13.5s]
2018-02-28T20:49:38.3016827Z 
2018-02-28T20:49:38.3016827Z Folder Capacity
2018-02-28T20:49:38.3016827Z Sub-folder api/Capacity
2018-02-28T20:49:39.9579504Z   GET http://test-services-TA.Humana.com/COSAPI/api/Capacity [200 OK, 4.06KB, 1609ms]
2018-02-28T20:49:39.9892009Z   Pass  GET api/Capacity - Response must be valid and have a body
2018-02-28T20:49:39.9892009Z   Pass  GET api/Capacity - Schema is valid
2018-02-28T20:49:40.0360759Z 
```

The tests which I am running are:

``` javascript
var schema
var data

try {
// Test for valid schema
schema = JSON.parse(pm.globals.get("capacityArraySchema"));

pm.test('GET api/Capacity - Schema is valid', function() {
    data = pm.response.json();
    var results = tv4.validateMultiple(data, schema);

    pm.expect(results.valid).to.be.true;
});
} 
catch (err) {
    console.log(err);
    console.log(schema);
    console.log(data);
    console.log(pm.globals.get("capacityArraySchema"));
    // console.log(pm.response);
}
```
And the output in text form:
```
? Capacity
└ api/Capacity
  GET http://localhost:12711/api/Capacity [200 OK, 4.2KB, 39ms]
  √  GET api/Capacity - Response must be valid and have a body
  ┌
  │ { type: 'Error',
  │   name: 'JSONError',
  │   message: 'Unexpected token u in JSON at position 0'
  │ }
  │ null
  │ {}
  │ null
  └
```

The capacityArraySchema in the globals file, which is exported from Postman, is:

```json
    {
      "key": "capacityArraySchema",
      "value": "{\n  \"type\": \"array\",\n  \"$schema\": \"http://json-schema.org/draft-04/schema#\",\n  \"description\": \"\",\n  \"minItems\": 1,\n  \"uniqueItems\": true,\n  \"items\": {\n    \"type\": \"object\",\n    \"required\": [\n      \"CapacityID\",\n      \"RegionID\",\n      \"IsSeniorCareCoach\",\n      \"IsFrontLineLeader\",\n      \"IsCareCoach\",\n      \"Capacity1\"\n    ],\n    \"properties\": {\n      \"CapacityID\": {\n        \"type\": \"number\"\n      },\n      \"RegionID\": {\n        \"type\": \"number\"\n      },\n      \"IsSeniorCareCoach\": {\n        \"type\": \"boolean\"\n      },\n      \"IsFrontLineLeader\": {\n        \"type\": \"boolean\"\n      },\n      \"IsCareCoach\": {\n        \"type\": \"boolean\"\n      },\n      \"Capacity1\": {\n        \"type\": \"number\",\n        \"minimum\": 0\n      }\n    }\n  }\n}",
      "enabled": true,
      "type": "text"
    }
```

I have also run the newman test with the globals file which I downloaded from TFS, and it fails as well with the same error. So it's not the globals file that is exported causing the problem.

Thoughts?

**Comment by kunagpal** [comment]
@patrickgardella The option to specify global variable files is `--globals`, or `-g` for short. More details can be found here: https://github.com/postmanlabs/newman#newman-run-collection-file-source-options

Could you confirm that the `newman run` command works with `--globals` for you?

**Comment by patrickgardella** [comment]
@kunagpal Thanks for that.  Yes, it works with the `--globals` option, but gives the same result.

I had switched it to `--global` as a test step since the Postman website shows _that_ as the correct command: https://www.getpostman.com/docs/v6/postman/collection_runs/command_line_integration_with_newman 

`-g, --global [file]             Specify a Postman globals file as JSON [file]`

**Comment by kunagpal** [comment]
@patrickgardella Thanks for mentioning the point about the docs, it is being corrected here: https://github.com/postmanlabs/postman-docs/pull/1005. As for your issue, could you add the following statement to the very beginning of your test script and confirm that the variables are not being resolved?
```js
console.log(pm.globals.toObject());
```

Additionally, could you share the globals file here? (feel free to replace the actual variable values with dummy content)

**Comment by patrickgardella** [comment]
They are not in Newman (but are in Postman) with the fixed `--globals` option.  The empty `{}` before the error is the output of the pm.globals.toObject() log line:

```
? Capacity
└ api/Capacity
  GET http://localhost:12711/api/Capacity [200 OK, 4.2KB, 26ms]
  √  GET api/Capacity - Response must be valid and have a body
  ┌
  │ {}
  │ { type: 'Error',
  │   name: 'JSONError',
  │   message: 'Unexpected token u in JSON at position 0'
  │ }
  │ null
  │ {}
  │ null
  └
```

**BUT** I did find the problem when I was triple checking things!  Previously, I could put the collection file file at the end of the command line, it worked.

Now, whatever is following `newman run` is ignored unless it is the collection file. So if I move the globals to the end of the command line arguments, it fails to load the environment variables, since that's immediately following `newman run`:

```
newman run --environment tests\PostmanEnvironments\VisualStudio.postman_environment.json --color --insecure --reporters cli,junit --reporter-junit-export results\junit.xml --globals tests\PostmanEnvironments\globals.postman_globals.json tests/CareOptimizationServices.postman_collection.json --bail
```

But if the collection file is listed right after `newman run` it works fine:

```
newman run tests\CareOptimizationServices.postman_collection.json --environment tests\PostmanEnvironments\VisualStudio.postman_environment.json --color --insecure --reporters cli,junit --reporter-junit-export results\junit.xml --globals tests\PostmanEnvironments\globals.postman_globals.json --bail
```

So it seems like it is not throwing an error or warning if the collection file is out of place.

**Comment by kunagpal** [comment]
@patrickgardella Again, thanks for pointing that out, a similar discussion has happened here: https://github.com/postmanlabs/newman/issues/1336

TL;DR: The collection path is meant to the first argument after the run command.

**Comment by patrickgardella** [comment]
Thanks, @kunagpal. You are correct, and the documentation is explicit. 

The fallback mechanism that was discussed [here](https://github.com/postmanlabs/newman/issues/1336#issuecomment-350796642), allows for the collection to be at the end, but @deepakpathania comment said "but the options specified before the required collection path wouldn't take effect."

What I found shows that newman doesn't follow that. All of the command line options do take effect, _except_ for the first one listed. It would certainly be clearer if there was an explicit warning that the collection should be first **or** if _all_ of the options specified before the required collection path were ignored.  In either of those two cases, the user would quickly realize that the order of the command line arguments is wrong. Ignoring the first one makes it non-intuitive (which is very subjective, I know!).
